lib/kubernetes-deploy/runner.rb

```diff
@@ -283,12 +279,15 @@ def validate_namespace
       KubernetesDeploy.logger.info("Namespace #{@namespace} validated")
     end
 
-    def run_kubectl(*args, namespaced: true)
+    def run_kubectl(*args, namespaced: true, with_context: true)
       args = args.unshift("kubectl")
       if namespaced
-        raise FatalDeploymentError, "Namespace missing for namespaced command" unless @namespace
```

Comment:
Because it's been validated before and I felt like we don't need to duplicate it. But I see your point and we can leave the duplicated check if you strongly feel that it's necessary to check it twice (in case someone forgets extra equal sign in `if @context = "value"`